src/scenes/Organizations/Create/CreateOrganization.tsx

```diff
@@ -0,0 +1,53 @@
+import { IconButton } from '@material-ui/core';
+import { Close } from '@material-ui/icons';
+import { FORM_ERROR } from 'final-form';
+import { useSnackbar } from 'notistack';
+import React, { useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { Except } from 'type-fest';
+import { handleFormError, useCreateOrganizationMutation } from '../../../api';
+import { useSession } from '../../../components/Session';
+import {
+  CreateOrganizationForm,
+  CreateOrganizationFormProps as Props,
+} from './CreateOrganizationForm';
+
+export const CreateOrganization = (props: Except<Props, 'onSubmit'>) => {
+  const [createOrg] = useCreateOrganizationMutation();
+  const { enqueueSnackbar, closeSnackbar } = useSnackbar();
+  const navigate = useNavigate();
+  const [session, sessionLoading] = useSession();
+
+  useEffect(() => {
+    if (!sessionLoading && !session) {
+      navigate('/login');
+    }
+  }, [navigate, session, sessionLoading]);
+
+  const submit: Props['onSubmit'] = async (input) => {
+    try {
+      await createOrg({
+        variables: { input },
+      });
+      enqueueSnackbar(
+        `Successfully created org with name: ${input.organization.name}`,
+        {
+          variant: 'success',
+          action: (key: string) => (
+            <IconButton color="inherit" onClick={() => closeSnackbar(key)}>
+              <Close />
+            </IconButton>
+          ),
+        }
+      );
+    } catch (e) {
+      return await handleFormError(e, {
+        Default: {
+          [FORM_ERROR]: `Something wasn't right. Try again.`,
+        },
+      });
```

Comment:
Error handlers can just return a string to indicate the same thing.
```diff
- Default: {
-   [FORM_ERROR]: `Something wasn't right. Try again.`,
- },
+ Default: `Something wasn't right. Try again.`,
```
However, I'm also thinking we should probably not copy/paste this generic error string everywhere. Since we aren't handling a specific case here let's just leave it off. Currently that means that the server message will be used. Later if we want to replace that with a generic message we can do that in the `handleFormErrors` function.
```suggestion
      return await handleFormError(e);
```

I know this is copied from the pattern in Authentication. It's a little unfortunate that we started with that, since because of the nature of it, it breaks the rules.

Reply:
ok, got it.